app/models/work.rb

```diff
@@ -120,20 +120,30 @@ def publish_test_doi?
 
     private
 
-      def works_by_user_state(user, state)
-        # The user's own works by state (if any)
-        works = Work.where(created_by_user_id: user, state: state).to_a
+      def search_terms_where_clause(search_terms)
+        if search_terms.nil?
+          Work.where("true")
+        else
+          Work.where("CAST(metadata AS VARCHAR) ilike :search_terms", search_terms: "%" + search_terms.strip + "%")
+        end
+      end
+
+      def works_by_user_state(user, state, search_terms)
+        search_terms_filter = search_terms_where_clause(search_terms)
+
+        # The user's own works (if any) by state and search terms
+        works = Work.where(created_by_user_id: user, state: state).and(search_terms_filter).to_a
```

Comment:
I had never chained WHERE clauses like this but it became handy for this case.

Reply:
I have always chained the queries using a memo, but this is a nice way too: https://github.com/pulibrary/lockers_and_study_spaces/blob/17260cee97525c04e955d0a50f6e898aba30af42/app/models/locker_assignment.rb#L18